### PR TITLE
prov/gni: Implement DGRAM endpoint type

### LINKED ISSUE #638: Implement DGRAM endpoint type
DGRAM endpoints are similar to RDM except not reliable.  With RDM endpoints mostly done, it may be easy to support DGRAM EPs for all the same functionality.

Should we disable any internal retransmission attempts when using DGRAM vs RDM?

### PR DESCRIPTION
**Pull request by chuckfossen**
The general strategy is to piggyback on the RDM endpoint type as much as
possible.

Changes to rdm_rma.c are the start of testing and are not complete.

Fixes #638

Signed-off-by: Chuck Fossen chuckf@cray.com

@sungeunchoi @ztiffany @hppritcha

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

File: prov/gni/test/rdm_rma.c [hunk 1]
```diff
@@ -58,7 +58,7 @@
 
 #include <criterion/criterion.h>
 
-#if 1
+#if 0
```
Comment (ztiffany) [inline prov/gni/test/rdm_rma.c]: Not for checkin.

**Comment by hppritcha** [comment]
:+1: modul the `if 0`

**Comment by ztiffany** [comment]
:+1: The test changes turned out great.

**Pull request merged**